src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -102,40 +180,55 @@
           [content.deleted/deleted-message message-data context]
           [message/message-with-reactions message-data context keyboard-shown])]))])
 
-(defn messages-list
-  [{:keys [chat-id] :as chat}]
-  (let [context        (rf/sub [:chats/current-chat-message-list-view-context])
-        messages       (rf/sub [:chats/raw-chat-messages-stream chat-id])
-        keyboard-shown (atom false)
-        bottom-space   15]
-    [rn/view
-     {:style {:flex 1}}
-     ;; NOTE: DO NOT use anonymous functions for handlers
-     [rn/flat-list
+(defn scroll-handler
+  [event initial-y scroll-y]
+  (let [content-size-y (- (oops/oget event "nativeEvent.contentSize.height")
+                          (oops/oget event "nativeEvent.layoutMeasurement.height"))
+        current-y      (+ (oops/oget event "nativeEvent.contentOffset.y") initial-y)]
+    (reanimated/set-shared-value scroll-y (- content-size-y current-y))))
+
+(defn messages-view
+  [{:keys [chat insets initial-y scroll-y cover-bg-color]}]
+  (let [render-data (rf/sub [:chats/current-chat-message-list-view-context])
+        messages    (rf/sub [:chats/raw-chat-messages-stream (:chat-id chat)])]
+    [rn/view {:style {:flex 1}}
+     [reanimated/flat-list
```

Comment:
Actually the current `flat-list` component in `reanimated` namespace is not an Animated flatlist. It need to be `(def flat-list (reagent/adapt-react-class (.-FlatList reanimated)))`. But this currently causes unit tests to fail.